build/docs-prep.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+# Function to print colored messages
+print_success() {
+  echo -e "${GREEN}✓ $1${NC}"
+}
+
+print_error() {
+  echo -e "${RED}✗ $1${NC}"
+  exit 1
+}
+
+print_info() {
+  echo -e "${BLUE}ℹ $1${NC}"
+}
+
+print_warning() {
+  echo -e "${YELLOW}⚠ $1${NC}"
+}
+
+# Function to execute command with error handling
+execute() {
+  print_info "Running: $1"
+  eval $1
+  if [ $? -ne 0 ]; then
+    print_error "Failed to execute: $1"
+  else
+    print_success "Successfully executed: $1"
+  fi
+}
+
+# Check if /tmp/_site directory exists from a previous run
+if [ -d "/tmp/_site" ]; then
+  print_warning "Found existing /tmp/_site directory. Removing it..."
+  rm -rf /tmp/_site
+fi
+
+# Main process
+print_info "Starting documentation deployment process..."
+
+# Step 1: Build documentation
+print_info "Building documentation with npm run docs..."
+npm run docs
+if [ $? -ne 0 ]; then
+  print_error "Documentation build failed!"
+fi
+print_success "Documentation built successfully"
+
+# Step 2: Move _site to /tmp/
+print_info "Moving _site to temporary location..."
+execute "mv _site /tmp/"
+
+# Step 3: Switch to gh-pages branch
+print_info "Checking out gh-pages branch..."
+git checkout gh-pages
+if [ $? -ne 0 ]; then
+  print_error "Failed to checkout gh-pages branch. Make sure it exists."
+fi
+print_success "Switched to gh-pages branch"
+
+# Step 4: Create a new branch for the update
+print_info "Creating new branch gh-pages-jd-docs-astro..."
+execute "git checkout -b gh-pages-jd-docs-astro"
```

Comment:
We might want to generate a unique name to avoid:

```
fatal: a branch named 'gh-pages-jd-docs-astro' already exists
✗ Failed to execute: git checkout -b gh-pages-jd-docs-astro
```